core/templates/pages/admin-page/features-tab/admin-features-tab.component.spec.ts

```diff
@@ -0,0 +1,372 @@
+
+      component.moveRuleUp(featureFlag, 1);
+      expect(featureFlag.rules[0].valueWhenMatched).toBeFalse();
+    });
+  });
+
+  describe('.moveRuleDown', () => {
+    it('should move rule down', () => {
+      const featureFlag = component.featureFlags[0];
+      component.addNewRuleToBottom(featureFlag);
+
+      component.moveRuleDown(featureFlag, 0);
+      expect(featureFlag.rules[1].valueWhenMatched).toBeTrue();
+    });
+  });
+
+  describe('.addNewFilter', () => {
+    it('should add new filter', () => {
+      const rule = component.featureFlags[0].rules[0];
+
+      expect(rule.filters.length).toBe(1);
+      component.addNewFilter(rule);
+      expect(rule.filters.length).toBe(2);
+    });
+  });
+
+  describe('.removeFilter', () => {
+    it('should remove filter', () => {
+      const rule = component.featureFlags[0].rules[0];
+
+      expect(rule.filters.length).toBe(1);
+      component.removeFilter(rule, 0);
+      expect(rule.filters.length).toBe(0);
+    });
+  });
+
+  describe('.addNewCondition', () => {
+    it('should add new condition', () => {
+      const filter = component.featureFlags[0].rules[0].filters[0];
+
+      expect(filter.conditions.length).toBe(1);
+      component.addNewCondition(filter);
+      expect(filter.conditions.length).toBe(2);
+    });
+  });
+
+  describe('.removeCondition', () => {
+    it('should remove condition', () => {
+      const filter = component.featureFlags[0].rules[0].filters[0];
+
+      expect(filter.conditions.length).toBe(1);
+      component.removeCondition(filter, 0);
+      expect(filter.conditions.length).toBe(0);
+    });
+  });
+
+
+  describe('.onFilterTypeSelectionChanged', () => {
+    it('should clear existing conditions', () => {
+      const filter = component.featureFlags[0].rules[0].filters[0];
+
+      expect(filter.conditions.length).toBe(1);
+      component.onFilterTypeSelectionChanged(filter);
+      expect(filter.conditions.length).toBe(0);
+    });
+  });
+
+  describe('.clearChanges', () => {
+    it('should clear changes', () => {
+      const featureFlag = component.featureFlags[0];
+
+      expect(featureFlag.rules.length).toBe(1);
+
+      component.addNewRuleToTop(featureFlag);
+      component.clearChanges(featureFlag);
+
+      expect(featureFlag.rules.length).toBe(1);
+    });
+
+    it('should not proceed if the user does\'t confirm', () => {
+      mockConfirmResult(false);
+      const featureFlag = component.featureFlags[0];
+
+      expect(featureFlag.rules.length).toBe(1);
+
+      component.addNewRuleToTop(featureFlag);
+      component.clearChanges(featureFlag);
+
+      expect(featureFlag.rules.length).toBe(2);
+    });
+  });
+
+  describe('.updateFeatureRulesAsync', () => {
+    let setStatusSpy: jasmine.Spy;
+
+    beforeEach(() => {
+      setStatusSpy = jasmine.createSpy();
+      component.setStatusMessage = setStatusSpy;
+
+      adminTaskManagerService.finishTask();
+      mockConfirmResult(true);
```

Comment:
For this & other tests above/below that use similar mock setups: I suggest always keeping the mock calls in relevant methods rather than defaulting it. Otherwise, it's hard to verify that the happy path tests are actually verifying the right thing since the reader needs to go searching for wherever the other part of the test arrangement is happening.